File: app/src/main/java/com/example/networkaplication/retrofit/OMDBApiInterface.java
package com.example.networkaplication.retrofit;

import com.example.networkaplication.models.search.Search;
import com.example.networkaplication.models.select.MovieDetail;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface OMDBApiInterface {
    String API_KEY = "?&apikey=c1645dc5";

    @GET(API_KEY)
    Call<Search> getSearchResult(@Query("s") String search);

    @GET(API_KEY)
    Call<MovieDetail> getMovieDetails(@Query("t") String movieName);
}

File: app/src/main/java/com/example/networkaplication/retrofit/OMDBApiInterface.kt
package com.example.networkaplication.retrofit

import com.example.networkaplication.models.search.Search
import com.example.networkaplication.models.select.MovieDetail

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface OMDBApiInterface {

    @GET("?&apikey=c1645dc5")
    fun getSearchResult(@Query("s") search: String): Call<Search>

    @GET("?&apikey=c1645dc5")
    fun getMovieDetails(@Query("t") movieName: String): Call<MovieDetail>
}
